packages/webcert/src/components/modal/CareProviderModal.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import FocusTrap from 'focus-trap-react'
+import { useDispatch, useSelector } from 'react-redux'
+import { getUser } from '../../store/user/userSelectors'
+import { setUnit } from '../../store/user/userActions'
+import { CareProviders } from './CareProviders'
+import { User } from '@frontend/common'
+
+const ModalContentWrapper = styled.div`
+  p + p {
+    margin-top: 0.25em !important;
+  }
+`
+
+const WrapText = styled.div`
+  white-space: normal;
+`
+
+interface Props {
+  title: string
+}
+
+const CareProviderModal: React.FC<Props> = ({ title }) => {
+  const dispatch = useDispatch()
+  const user = useSelector(getUser)
+  const [isOpen, setIsOpen] = useState(false)
+
+  const handleChooseUnit = (event: React.MouseEvent) => {
+    const unitId = event.currentTarget.id
+
+    dispatch(setUnit(unitId))
+    setIsOpen(false)
+  }
+
+  useEffect(() => {
+    if (user) {
+      if (user.loggedInUnit.unitId) {
+        return
+      }
+
+      setIsOpen(true)
+    }
+  }, [user, dispatch])
+
+  if (!isOpen || user?.careProviders.length === 0) {
+    return null
+  }
+
+  const { careProviders } = user as User
+
+  return (
+    <FocusTrap active={isOpen}>
```

Comment:
Can the toast modal be used instead and deactivate the automatic closing? Otherwise the modal can be extracted into a more generalized component (and placed in common) and the content be sent in as children.

Reply:
A new generic popup modal has been created.